dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1489,55 +1454,60 @@ protected void indexCount(final String query,
+                    RequestOptions.DEFAULT, new ActionListener<CountResponse>() {
+                        @Override
+                        public void onResponse(CountResponse countResponse) {
+                            indexCountSuccess.accept(countResponse.getCount());
+                        }
 
+                        @Override
+                        public void onFailure(Exception e) {
+                            if (null != indexCountFailure) {
+                                indexCountFailure.accept(e);
+                            }
+                        }
+                    });
+        }catch (Exception ee){
+            Logger.fatal(this, "Can't get indices information",ee);
+            if (null != indexCountFailure) {
                 indexCountFailure.accept(ee);
             }
-            return;
         }
 
-        SearchRequest searchRequest = getCountSearchRequest(queryStringQuery);
-        searchRequest.indices(query.contains("+live:true") && !query.contains("+deleted:true")?
-                info.getLive(): info.getWorking());
-
-        RestHighLevelClientProvider.getInstance().getClient().searchAsync(searchRequest, RequestOptions.DEFAULT,
-                        new ActionListener<SearchResponse>() {
-            @Override
-            public void onResponse(SearchResponse searchResponse) {
-
-                indexCountSuccess.accept(searchResponse.getHits().getTotalHits().value);
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-
-                if (null != indexCountFailure) {
-
-                    indexCountFailure.accept(e);
-                }
-            }
-        });
     }
 
     @NotNull
-    private SearchRequest getCountSearchRequest(final String queryString) {
-        SearchRequest searchRequest = new SearchRequest();
-        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-        searchSourceBuilder.query(QueryBuilders.queryStringQuery(queryString));
-        searchSourceBuilder.size(0);
-        searchSourceBuilder.timeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS));
-        searchRequest.source(searchSourceBuilder);
-        return searchRequest;
+    private CountRequest getCountRequest(final String queryString) {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.query(QueryBuilders.queryStringQuery(queryString));
+        final CountRequest countRequest = new CountRequest(inferIndexToHit(queryString));
+        countRequest.source(sourceBuilder);
+        return countRequest;
     }
 
+   private String inferIndexToHit(final String query)  {
+       // we check the query to figure out which indexes to hit
+
+       final IndiciesInfo info;
+       try {
+           info = APILocator.getIndiciesAPI().loadIndicies();
+       } catch (DotDataException e) {
+           throw new DotRuntimeException(e);
+       }
+
+       final String indexToHit;
```

Comment:
I would change this to

return query.contains("+live:true") && !query.contains("+deleted:true")? info.getLive(): info.getWorking()